systems/controllers/cp_traj_gen.cc

```diff
@@ -156,28 +147,16 @@ void CPTrajGenerator::calcCpAndStanceFootHeight(
       (BasicVector<double>*)this->EvalVectorInput(context, fsm_port_);
   VectorXd fsm_state = fsm_output->get_value();
 
-  // Get stance foot position and index
-  KinematicsCache<double> cache = tree_.CreateKinematicsCache();
   VectorXd q = robot_output->GetPositions();
-  // Modify the quaternion in the begining when the state is not received from
-  // the robot yet
-  if (is_quaternion_) {
-    multibody::SetZeroQuaternionToIdentity(&q);
-  }
-  cache.initialize(q);
-  tree_.doKinematics(cache);
-
-  int stance_foot_idx;
-  Vector3d pt_on_stance_foot;
-  if (fsm_state(0) == left_right_support_fsm_states_[1]) {
-    stance_foot_idx = left_right_foot_body_indices_[1];
-    pt_on_stance_foot = left_right_foot_pts_on_bodies_[1];
-  } else {
-    stance_foot_idx = left_right_foot_body_indices_[0];
-    pt_on_stance_foot = left_right_foot_pts_on_bodies_[0];
-  }
-  Vector3d stance_foot_pos =
-      tree_.transformPoints(cache, pt_on_stance_foot, stance_foot_idx, 0);
+  plant_.SetPositions(context_.get(), q);
+
+  // Stance foot position
+  auto stance_foot = (fsm_state(0) == left_right_support_fsm_states_[1])
```

Comment:
this should be a map from the fsm_state to the stance_foot

Reply:
Done